gio-subclass/src/application.rs

```diff
@@ -0,0 +1,655 @@
+use gio;
+use gio_ffi;
+use glib;
+use glib::translate::*;
+use glib::IsA;
+use glib_ffi;
+use gobject_ffi;
+use libc;
+use std::mem;
+use std::ptr;
+use std::ops::Deref;
+use std::fmt;
+use std::convert;
+
+use gobject_subclass::anyimpl::*;
+use gobject_subclass::object::*;
+
+pub struct ArgumentList{
+    pub(crate) ptr: *mut *mut *mut libc::c_char,
+    items: Vec<String>
+}
+
+impl ArgumentList{
+
+    pub(crate) fn new(arguments: *mut *mut *mut libc::c_char) -> Self{
+        Self{
+            ptr: arguments,
+            items: unsafe{
+                FromGlibPtrContainer::from_glib_none(ptr::read(arguments))
+            }
+        }
+    }
+
+    // remove the item at index `idx` and shift the raw array
+    pub fn remove(&mut self, idx: usize){
+
+        unsafe{
```

Comment:
Unneeded empty line and missing space before the {. Maybe just run everything once through rustfmt-nightly (just get the version via rustup)